viewer/src/public/RevealManagerBase.ts

```diff
@@ -58,110 +29,49 @@ export type OnDataUpdated = () => void;
 
 export class RevealManagerBase {
   protected readonly _materialManager: MaterialManager;
-  private readonly _sectorRepository: Repository;
-  private readonly _modelSubject: Subject<Cad | PointCloud> = new Subject();
-  private readonly _loadingHintsSubject: Subject<CadLoadingHints> = new BehaviorSubject({});
-  private readonly _cameraSubject: Subject<THREE.PerspectiveCamera>;
-  private readonly _modelObservable: Observable<CadNode | [PotreeGroupWrapper, PotreeNodeWrapper]>;
-  private readonly _cadModelObservable: Observable<CadModel[]>;
+  protected readonly _cadManager: CadManager;
 
-  private readonly _cadNodeMap: Map<string, CadNode> = new Map();
+  private _client: CogniteClient;
 
-  private _sectorCuller: SectorCuller;
-  // private _budget: CadBudget;
-
-  constructor(
-    client: CogniteClient,
-    sectorRepository: Repository,
-    materialManager: MaterialManager,
-    dataUpdatedCallback: OnDataUpdated,
-    options?: RevealOptions
-  ) {
-    this._sectorCuller = (options && options.internal && options.internal.sectorCuller) || new ProximitySectorCuller();
+  constructor(client: CogniteClient, cadManager: CadManager, materialManager: MaterialManager) {
     // this._budget = (options && options.budget) || createDefaultCadBudget();
-    this._sectorRepository = sectorRepository;
+    this._client = client;
+    this._cadManager = cadManager;
     this._materialManager = materialManager;
-    this._modelObservable = this._modelSubject.pipe(this.addCadOrPointcloudModel(client));
-    this._cadModelObservable = this._modelObservable.pipe(this.getAddedModels());
-    this._cameraSubject = this.createLoadSectorsPipeline(dataUpdatedCallback);
+  }
+
+  public resetRedraw(): void {
+    this._cadManager.resetRedraw();
+  }
```

Comment:
Do we need to have a `resetRedraw`-function or can we do this using the `Mesh`-trick I showed? E.g. can we provide a new field on CadNode called lastRenderTimestamp and compare that with a timestamp of the last event that caused needsRedraw to be set?

Reply:
I did not like the Mesh trick due to it having to be added to scene, and I didn't just want to add it without having a proper solution for it. ie (we have 1 mesh for the entire manager, and we make sure that it follows the camera (due to well, if not can't this be outside the frustum and not rendered at some point?)

Reply:
`onBeforeRender` is called before frustum culling I believe, and if not it's possible to disable frustum culling for an object (https://threejs.org/docs/#api/en/core/Object3D.frustumCulled).

I agree that adding an empty mesh to trigger the callback is code smell, but I think it's worth it as `resetRedraw` is prone to user errors.